lib/core/versions/latest/OperationProcessor.ts

```diff
@@ -163,26 +163,24 @@ export default class OperationProcessor implements IOperationProcessor {
       return didState;
     };
 
+    // Passed all verifications, must update the update commitment value even if the application of patches fail.
+    const newDidState = {
+      nextRecoveryCommitmentHash: didState.nextRecoveryCommitmentHash,
+      document: didState.document,
+      nextUpdateCommitmentHash: operation.delta.updateCommitment,
+      lastOperationTransactionNumber: anchoredOperationModel.transactionNumber
+    };
+
     let resultingDocument;
     try {
       resultingDocument = await DocumentComposer.applyUpdateOperation(operation, didState.document);
```

Comment:
I think the second argument needs to be a deep copy, something like Object.assign({}, didState.docuemnt) should work.

Reply:
Thanks for the great catch!